src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -540,61 +540,72 @@ bool Connection::websocket_closed_handler(bool was_clean, Status status)
         case WebSocketError::websocket_no_status_received:
             [[fallthrough]];
         case WebSocketError::websocket_invalid_extension: {
-            constexpr bool try_again = true;
-            SessionErrorInfo error_info{std::move(status), try_again};
-            involuntary_disconnect(std::move(error_info),
-                                   ConnectionTerminationReason::websocket_protocol_violation); // Throws
+            close_due_to_client_side_error({ErrorCodes::SyncProtocolInvariantFailed, msg}, IsFatal{false},
```

Comment:
should we try to get rid off of calling `involuntary_disconnect` directly and call `close_due_to_...` methods instead? I see you fixed it here.

Reply:
I don't know? The difference between them right now is that close_due_to_ constructs a SessionErrorInfo for you. I'd almost go for getting rid of close_due_to_ methods in favor of involuntary_disconnect instead? But I don't feel very strongly either way.

Reply:
Right. It could be `involuntary_disconnect` instead, but I don't really like having both.

Reply:
Okay, can I do that in a separate PR or do you need to see it done in this one?

Reply:
not really now. It would be nice at some point.